Pod/Classes/NSAttributedString+BonMotUtilities.m

```diff
@@ -0,0 +1,88 @@
+//
+//  NSAttributedString+BonMotUtilities.m
+//  Pods
+//
+//  Created by Eliot Williams on 3/9/16.
+//
+//
+
+#import "NSAttributedString+BonMotUtilities.h"
+#import "BONChain.h"
+#import "BONSpecial.h"
+
+@implementation NSAttributedString (BonMotUtilities)
+
+- (NSString *)humanReadableString
+{
+    NSString *originalString = self.string;
+    NSMutableString *composedHumanReadableString = [NSMutableString string];
+
+    [originalString enumerateSubstringsInRange:NSMakeRange(0, originalString.length) options:NSStringEnumerationByComposedCharacterSequences usingBlock:^(NSString *BONCNullable substring, NSRange substringRange, NSRange enclosingRange, BOOL *BONCNonnull stop) {
+
+        static NSCharacterSet *s_newLineCharacterSet = nil;
+        if (!s_newLineCharacterSet) {
+            s_newLineCharacterSet = [NSCharacterSet newlineCharacterSet];
+        }
+        NSCharacterSet *s_whiteSpaceAndNewLinesSet = nil;
+        if (!s_whiteSpaceAndNewLinesSet) {
+            s_whiteSpaceAndNewLinesSet = [NSCharacterSet whitespaceAndNewlineCharacterSet];
+        }
+
+        // Substitute Images with @"{image#heightx#width}"
+        if ([substring isEqualToString:BONSpecial.objectReplacementCharacter]) {
+            BONStringDict *attributes = [self attributesAtIndex:substringRange.location effectiveRange:NULL];
+            NSTextAttachment *attachment = attributes[NSAttachmentAttributeName];
+            UIImage *attachedImage = attachment.image;
+            NSString *imageSubstitutionString = [NSString stringWithFormat:@"{image%.0fx%.0f}", attachedImage.size.height, attachedImage.size.width];
+            [composedHumanReadableString appendString:imageSubstitutionString];
+        }
+        else {
+            // Substitute BONSpecial Characters with @"{#name}"
+            unichar character = [substring characterAtIndex:0];
+            BONGeneric(NSDictionary, NSNumber *, NSString *)*specialNames = @{
+                @(BONCharacterLineFeed) : @"{lineFeed}",
```

Comment:
#? any way to auto-generate this list using `BONSpecialGenerator.swift`, so that it will be updated when new characters are added to `BONSpecial`? Happy to discuss options here, since the generator script has some weird edge cases.